### PR TITLE
(feat) O3-3207: Navigate to appointments by day from appointments calendar 

### PR DESCRIPTION
**Pull request by usamaidrsk**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

PR adds that one can navigate to appointments of a given day, when clicked on from the appointments calendar

## Screenshots

https://github.com/openmrs/openmrs-esm-patient-management/assets/53287480/70139dda-bb52-4fc0-92ae-c752d755ec9f

## Other

[O3-3207](https://openmrs.atlassian.net/browse/O3-3207)

[O3-3207]: https://openmrs.atlassian.net/browse/O3-3207?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/esm-appointments-app/src/calendar/monthly/monthly-workload-view.component.tsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ const MonthlyWorkloadView: React.FC<MonthlyWorkloadViewProps> = ({ dateTime, eve
       )}>
       {isSameMonth(dateTime, dayjs(selectedDate)) && (
         <p>
-          <div className={classNames(styles.totals)}>
+          <div className={classNames(styles.totals)} onClick={() => onClick('')}>
```
Comment (mogoodrich) [inline packages/esm-appointments-app/src/calendar/monthly/monthly-workload-view.component.tsx:64]: interesting... how does this work @usamaidrsk ?

Comment (usamaidrsk) [reply packages/esm-appointments-app/src/calendar/monthly/monthly-workload-view.component.tsx:64]:
```typescript
const onClick = (serviceUuid: string) => {
    navigate({ to: `${spaHomePage}/appointments/${dayjs(dateTime).format('YYYY-MM-DD')}/${serviceUuid}` });
  };
``` 

This here is the function definition @mogoodrich .
Which is also used on here.
```typescript
{visibleServices.map(({ serviceName, serviceUuid, count }, i) => (
                <div
                  key={`${serviceUuid}-${count}-${i}`}
                  role="button"
                  tabIndex={0}
                  onClick={(e) => {
                    e.stopPropagation();
                    onClick(serviceUuid);
                  }}
                  className={styles.serviceArea}>
                  <span>{serviceName}</span>
                  <span>{count}</span>
                </div>
              ))}
```

Comment (usamaidrsk) [reply packages/esm-appointments-app/src/calendar/monthly/monthly-workload-view.component.tsx:64]: Maybe I need to rename the function to be more meaningful from what was there 
Thanks

Comment (mogoodrich) [reply packages/esm-appointments-app/src/calendar/monthly/monthly-workload-view.component.tsx]: Oh, nevermind, I get it now, thanks!

**Review by mogoodrich — APPROVED** [review]
Looks good, thanks @usamaidrsk , I will merge!

**Pull request merged**